File: android/src/main/java/com/biso/capacitor/plugins/mlkit/barcode/scanner/BarcodeType.java
package com.biso.capacitor.plugins.mlkit.barcode.scanner;

import com.google.mlkit.vision.barcode.common.Barcode;

public enum BarcodeType {
  UNKNOWN(Barcode.TYPE_UNKNOWN),
  CONTACT_INFO(Barcode.TYPE_CONTACT_INFO),
  EMAIL(Barcode.TYPE_EMAIL),
  ISBN(Barcode.TYPE_ISBN),
  PHONE(Barcode.TYPE_PHONE),
  PRODUCT(Barcode.TYPE_PRODUCT),
  SMS(Barcode.TYPE_SMS),
  TEXT(Barcode.TYPE_TEXT),
  URL(Barcode.TYPE_URL),
  WIFI(Barcode.TYPE_WIFI),
  GEO(Barcode.TYPE_GEO),
  CALENDAR_EVENT(Barcode.TYPE_CALENDAR_EVENT),
  DRIVER_LICENSE(Barcode.TYPE_DRIVER_LICENSE);
  private final int type;

  BarcodeType(int type) {
    this.type = type;
  }

  public int getAsInt() {
    return type;
  }

  public static BarcodeType getFromInt(int intValue) {
    for (BarcodeType value : BarcodeType.values()) {
      if (value.type == intValue) {
        return value;
      }
    }
    return null;
  }
}

File: ios/Plugin/BarcodeType.swift
import MLKitBarcodeScanning

public enum BarcodeType: String, CaseIterable {
    // swiftlint:disable identifier_name
    case UNKNOWN
    case CONTACT_INFO
    case EMAIL
    case ISBN
    case PHONE
    case PRODUCT
    case SMS
    case TEXT
    case URL
    case WIFI
    case GEO
    case CALENDAR_EVENT
    case DRIVER_LICENSE
    // swiftlint:enable identifier_name

    var asInt: Int {
        switch self {
        case BarcodeType.UNKNOWN: return MLKitBarcodeScanning.BarcodeValueType.unknown.rawValue
        case BarcodeType.CONTACT_INFO: return MLKitBarcodeScanning.BarcodeValueType.contactInfo.rawValue
        case BarcodeType.EMAIL: return MLKitBarcodeScanning.BarcodeValueType.email.rawValue
        case BarcodeType.ISBN: return MLKitBarcodeScanning.BarcodeValueType.ISBN.rawValue
        case BarcodeType.PHONE: return MLKitBarcodeScanning.BarcodeValueType.phone.rawValue
        case BarcodeType.PRODUCT: return MLKitBarcodeScanning.BarcodeValueType.product.rawValue
        case BarcodeType.SMS: return MLKitBarcodeScanning.BarcodeValueType.SMS.rawValue
        case BarcodeType.TEXT: return MLKitBarcodeScanning.BarcodeValueType.text.rawValue
        case BarcodeType.URL: return MLKitBarcodeScanning.BarcodeValueType.URL.rawValue
        case BarcodeType.WIFI: return MLKitBarcodeScanning.BarcodeValueType.wiFi.rawValue
        case BarcodeType.GEO: return MLKitBarcodeScanning.BarcodeValueType.geographicCoordinates.rawValue
        case BarcodeType.CALENDAR_EVENT: return MLKitBarcodeScanning.BarcodeValueType.calendarEvent.rawValue
        case BarcodeType.DRIVER_LICENSE: return MLKitBarcodeScanning.BarcodeValueType.driversLicense.rawValue
        }
    }

    static func getFromInt(intValue: Int) -> BarcodeType? {
        for barcodeType in BarcodeType.allCases where barcodeType.asInt == intValue {
            return barcodeType
        }
        print("no BarcodeType found for value ", intValue)
        return nil
    }
}
